src/main/java/pageObjects/CreateDevicePage.java

```diff
@@ -0,0 +1,223 @@
+        devicesButton.click();
+        addDeviceButton.click();
+        nameField.sendKeys(DEVICE_NAME);
+        serialNoField.sendKeys(DEVICE_SERIAL_NO_WS);
+        inventoryNoField.sendKeys(DEVICE_INVENTORY_NO);
+        invoiceNoField.sendKeys(DEVICE_INVOICE_NO);
+        saveDeviceButton.click();
+    }
+
+    public void createDeviceWithWhiteSpaceInventoryNoField() {
+        devicesButton.click();
+        addDeviceButton.click();
+        nameField.sendKeys(DEVICE_NAME);
+        serialNoField.sendKeys(DEVICE_SERIAL_NO);
+        inventoryNoField.sendKeys(DEVICE_INVENTORY_NO_WS);
+        invoiceNoField.sendKeys(DEVICE_INVOICE_NO);
+        saveDeviceButton.click();
+    }
+
+    public void createDeviceWithWhiteSpaceInvoiceNoField() {
+        devicesButton.click();
+        addDeviceButton.click();
+        nameField.sendKeys(DEVICE_NAME);
+        serialNoField.sendKeys(DEVICE_SERIAL_NO);
+        inventoryNoField.sendKeys(DEVICE_INVENTORY_NO);
+        invoiceNoField.sendKeys(DEVICE_INVOICE_NO_WS);
+        saveDeviceButton.click();
+    }
+
+    public void createDeviceWithThiteSpaceDescriptionField() {
+        devicesButton.click();
+        addDeviceButton.click();
+        nameField.sendKeys(DEVICE_NAME);
+        serialNoField.sendKeys(DEVICE_SERIAL_NO);
+        inventoryNoField.sendKeys(DEVICE_INVENTORY_NO);
+        invoiceNoField.sendKeys(DEVICE_INVOICE_NO);
+        descriptionField.sendKeys(DEVICE_DESCRIPTION_WS);
+        saveDeviceButton.click();
+    }
+
+    public void createDeviceTooLongName() {
+        devicesButton.click();
+        addDeviceButton.click();
+        nameField.sendKeys(DEVICE_NAME_TOO_LONG);
+        serialNoField.sendKeys(DEVICE_SERIAL_NO);
+        inventoryNoField.sendKeys(DEVICE_INVENTORY_NO);
+        invoiceNoField.sendKeys(DEVICE_INVOICE_NO);
+        saveDeviceButton.click();
+    }
+
+    public void createDeviceTooLongSerialNo(){
+        devicesButton.click();
+        addDeviceButton.click();
+        nameField.sendKeys(DEVICE_NAME);
+        serialNoField.sendKeys(DEVICE_NUMBER_OVER_30_CHAR);
+        inventoryNoField.sendKeys(DEVICE_INVENTORY_NO);
+        invoiceNoField.sendKeys(DEVICE_INVOICE_NO);
+        saveDeviceButton.click();
+    }
+
+    public void createDeviceToLongInventoryNo(){
+        devicesButton.click();
+        addDeviceButton.click();
+        nameField.sendKeys(DEVICE_NAME);
+        serialNoField.sendKeys(DEVICE_SERIAL_NO);
+        inventoryNoField.sendKeys(DEVICE_NUMBER_OVER_30_CHAR);
+        invoiceNoField.sendKeys(DEVICE_INVOICE_NO);
+        saveDeviceButton.click();
+    }
+
+    public void createDeviceToLongInvoiceNo(){
+        devicesButton.click();
+        addDeviceButton.click();
+        nameField.sendKeys(DEVICE_NAME);
+        serialNoField.sendKeys(DEVICE_SERIAL_NO);
+        inventoryNoField.sendKeys(DEVICE_INVENTORY_NO);
+        invoiceNoField.sendKeys(DEVICE_NUMBER_OVER_30_CHAR);
+        saveDeviceButton.click();
+    }
+}
```

Comment:
are we able to create a device with empty date fields? (invoice date, warranty start/end). can we create devices using a end date earlier than start date?